src/PrinterCartridgeLevelAlert.php

```diff
@@ -0,0 +1,328 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+/**
+ *  PrinterCartridgeLevelAlert class
+ *
+ * @ since version 0.84
+ **/
+class PrinterCartridgeLevelAlert extends CommonGLPI
+{
+    protected static $notable = true;
+
+    public static function getTypeName($nb = 0)
+    {
+        return __('Printer Cartridge Level Alerts');
+    }
+
+    /**
+     * @since 0.85
+     **/
+    public static function canView()
+    {
+        return (Session::haveRight('printer', READ) && Session::haveRight('cartridge', READ));
+    }
+
+    /**
+     * Print a good title
+     *
+     *@return void
+     **/
+    public static function title()
+    {
+        Html::displayTitle(
+            "",
+            self::getTypeName(),
+            "<i class='fas fa-check fa-lg me-2'></i>" . self::getTypeName()
+        );
+    }
+
+    /**
+    * @param $entities
+    *        $repeat for already notified management
+    *
+    * @return string
+    */
+    private static function query($entities, $repeat = null)
+    {
+        global $DB;
+
+        $query = "SELECT c.id as cartridge, i.id as cartridgeitem, p.id as printer, p.entities_id as entity, l.value as cartridgelevel, a.id as alertID, a.date as alertDate
```

Comment:
No raw queries please. This method could return an array of criteria to use with `$DB->request` instead.

Reply:
I need some help. I was unable to correctly create a full array SQL construction.

Reply:
You can find documentaiton at: https://glpi-developer-documentation.readthedocs.io/en/master/devapi/database/dbiterator.html

Reply:
> You can find documentaiton at: https://glpi-developer-documentation.readthedocs.io/en/master/devapi/database/dbiterator.html

My concern is about extra characters like quotes that the parser adds by itself and that breaks sql syntax. Will try to take a look once again...